Datadog/Example/AppConfiguration.swift

```diff
@@ -6,6 +6,7 @@
 
 import UIKit
 import Datadog
+import DatadogInternal
```

Comment:
Why do we need to import `DatadogInternal` in application code? In theory, `DatadogInternal` shouldn't be used by end-users. Is this required by migration some step or Example project specifics?

Reply:
Ah good catch! we need to expose some types from `DatadogInternal`, but this should be transparent for the user.
We can either use wrapper types or use the `@_exported import` for some types. That second option is easier!

Reply:
> we need to expose some types from DatadogInternal, but this should be transparent for the user.

Which ones for example? Yes, I think `@_exported import` in our `Example` project should do the job 👍.

Reply:
Currently, I see the following (it needs to be defined in `Datadog` module)

```swift
@_exported import enum DatadogInternal.DatadogSite
@_exported import struct DatadogInternal.AttributeKey
@_exported import protocol DatadogInternal.AttributeValue
@_exported import enum DatadogInternal.TrackingConsent
```

**Note:** I'm thinking to remove `AttributeKey` and `AttributeValue` aliases at some point

Reply:
Ok, now it makes sense 👌 - we need these symbols in `Example` app because we add custom logics around it (e.g. use it in variables declaration). Such use case won't happen in customer app, because the symbol will be visible through `Datadog` or other Feature modules.

> Note: I'm thinking to remove AttributeKey and AttributeValue aliases at some point

Their role today is to [describe with code comments](https://github.com/DataDog/dd-sdk-ios/blob/5491d29171e08c939ee09cc0d6e51ad66309daa6/Sources/Datadog/Core/Attributes/Attributes.swift#L9-L63) how keys and values work across all products. If we remove these aliases, where would we describe it? The only place would be next to each method that depends on attributes, which would be tedious. Not sure why we'd want to remove them?

Reply:
Yes, the doc well explains it, but it is necessary because `[AttributeKey: AttributeValue]` is kindof confusing. While `[String: Encodable]` is clearer IMO. We could use a typealias like 
```swift
typealias DatadogAdditionalAttributes = [String: Encodable]
```

Reply:
`DatadogAdditionalAttributes = [String: Encodable]` could work as well 👍. I like type aliases because they have some benefits of extra types without requiring us to introduce new `structs` in fact. Let's keep this for later, agree 👍.